src/bin/server/worker.rs

```diff
@@ -322,7 +329,16 @@ impl Worker {
             }
         }
 
-        self.index.save(&self.index_file)?;
+        // To avoid persisting the index too many times to storage, we only persist it after some
+        // time elapsed since the last save.
+        match self.last_index_backup {
+            Some(last) if last.elapsed() >= MINIMUM_DELAY_BETWEEN_INDEX_BACKUPS => {
+                self.index.save(&self.index_file)?;
+                self.last_index_backup = Some(Instant::now());
```

Comment:
It might be a good idea to set this *before* we call the save(...) so that if we fail we're not suddenly persisting even more often. I don't think we need to guarantee hourly saves, if we go days without saving that's also mostly fine.

(Ideally we'd have an alarm on the S3 file's modification date or so).

Reply:
Done!